container-search/src/main/javacc/com/yahoo/search/grouping/request/parser/GroupingParser.jj

```diff
@@ -35,8 +35,14 @@ public class GroupingParser {
         if (str.indexOf('.') >= 0) {
             return Double.valueOf(str);
         }
+        if (str.equals(Long.toString(Long.MIN_VALUE).substring(1))) return Double.NaN;
         return Long.valueOf(str);
     }
+
+    private static GroupingExpression resolveNegation(GroupingExpression exp) {
+        if (exp instanceof DoubleValue && ((DoubleValue)exp).getValue().isNaN()) return new LongValue(Long.MIN_VALUE);
```

Comment:
`-NaN`? 🙈

Reply:
Hehe I guess we don't parse `NaN` ... looking at the above. Ok, this is the dirtiest PR I'll ever approve 🤣

Reply:
My original efforts all stranded with left-recursion which JavaCC does not support. So it ended up with this little hack sadly.